tools.go

```diff
@@ -2,15 +2,48 @@ package loge
 
 import (
 	"fmt"
+	"os"
 	"path/filepath"
+	"sort"
+	"strconv"
 	"time"
 )
 
 const dateTimeStringLength = 27
 
+const maxFileSize = 5000000000
+
 func getLogName(path string) string {
 	t := time.Now()
-	ret := fmt.Sprintf("%d%02d%02d.log", t.Year(), t.Month(), t.Day())
+	ret := fmt.Sprintf("%d%02d%02d_", t.Year(), t.Month(), t.Day())
+	fileList, _ := os.ReadDir(path)
+	sort.Slice(fileList,
+		func (x int, y int) bool {
+			return fileList[x].Name() < fileList[y].Name()
+		})
+
+	fileNum := 0
+
+	if len(fileList) == 0 {
+		return filepath.Join(path, ret + fmt.Sprintf("%04d.log", fileNum))
+	}
+	
+	lastFileName := fileList[len(fileList) - 1].Name()
+	lastNum, _ := strconv.Atoi(lastFileName[9:13])
+	fileNum = lastNum
+
+	pathToFile := filepath.Join(path, ret + fmt.Sprintf("%04d.log", fileNum))
+	fi, err := os.Stat(pathToFile)
+	if err == nil {
+		if fi.Size() > maxFileSize {
+			fileNum += 1
+		}
+	}
```

Comment:
We may want to think about how we'd like to handle if there is an error. If we fail to stat the file, we might want to pass the error back to the caller of `getLogName`